mono/metadata/threads-types.h

```diff
@@ -342,12 +342,13 @@ typedef enum {
     MonoSetThreadNameFlag_Permanent = 0x0001,
     MonoSetThreadNameFlag_Reset     = 0x0002,
     MonoSetThreadNameFlag_Constant  = 0x0004,
+    MonoSetThreadNameFlag_RepeatedlyButOptimized = 0x0008,
```

Comment:
I don't like this name.
Some might shorten it to "Fast" (the "optimized" part, omitting the "repeatedly" part).
I don't like that either, popular but vague.
I couldn't come up with anything else.
It could also be, instead of the behavior, the user -- "ThreadPoolWorker".
But I just pushed past the analysis/naming paralysis. :)